### DISCUSSION TITLE
Get material for footsteps?

### DISCUSSION THREAD

**Discussion by elvisish** [blackears/cyclopsLevelBuilder#51] [Q&A] [answered]
Is it possible to get materials from blocks? I want to use different footsteps for different materials and but am not sure how to get them with a ray node (it just returns StaticBody3D or its children, CollisionShape3D).

**Comment by elvisish** [comment]
Okay this seems to sort of work, though it's a little clunky:
```swift
get_collider().get_parent().materials[0]
```
But this is just giving me a UID I think, instead of a reference to the material, or even the material name (most useful):
`<StandardMaterial3D#-9223371976825895526>`

**Reply by blackears** [reply]
Right now the mesh is stored in a MeshInstance3D which is inside a CyclopsConvexBlockBody, so you can get it the same way you do for a regular MeshInstance3D.

**Reply by elvisish** [reply]
> Right now the mesh is stored in a MeshInstance3D which is inside a CyclopsConvexBlockBody, so you can get it the same way you do for a regular MeshInstance3D.

Is this the regular way?
```swift
get_collider().get_parent().materials[0]
```
I could probably do `get_collider().get_parent().materials[0].name` and compare the string to lists of names that use particular sounds?

**Comment by blackears** [comment]
I'm not exactly sure what you're doing, but I think get_collider() returns the StaticBode3D with the collision.  The get_parent() takes you to the CyclopsConvexBlockBody which should have a child that is a MeshInstance3D which has the actual mesh.  You should be looking there for the material.  Then again, this is likely to change fairly shortly after the redesign.

**Reply by elvisish** [reply, marked as answer]
> I'm not exactly sure what you're doing, but I think get_collider() returns the StaticBode3D with the collision. The get_parent() takes you to the CyclopsConvexBlockBody which should have a child that is a MeshInstance3D which has the actual mesh. You should be looking there for the material. Then again, this is likely to change fairly shortly after the redesign.

Okay, here we go:
```swift
ray.get_collider().get_parent().materials[0].albedo_texture.resource_path.get_file().trim_suffix(".png")
```
I've no idea how performant this is, but it works. It would be nice if there was a slightly tidier way of doing this, but it's good enough for occasional footstep sounds.
